### PR TITLE
Make PCL compile with VTK 8

### PR DESCRIPTION
**Pull request by mvieth**
Tested with VTK 8.2 from homebrew on ubuntu
Behaviour for VTK 6, 7, and 9 is unchanged
@themightyoarfish Please test this

### CODE REVIEW

**Comment by mvieth** [comment]
@themightyoarfish Did you have time to test this yet?

**Comment by themightyoarfish** [comment]
Sorry not yet, but i keep it on unread so i don't forget to do it.

**Comment by themightyoarfish** [comment]
So on a clean build in Debug mode with vtk 8.2 on macos, I get at least one linker error. I think `vtkCommonColor` is missing somewhere.
```
[ 41%] Building CXX object filters/CMakeFiles/pcl_filters.dir/src/project_inliers.cpp.o
Undefined symbols for architecture x86_64:
  "vtkColorSeries::New()", referenced from:
      vtkSmartPointer<vtkColorSeries>::New() in pcl_plotter.cpp.o
  "vtkParametricEllipsoid::New()", referenced from:
      vtkSmartPointer<vtkParametricEllipsoid>::New() in shapes.cpp.o
  "vtkColorSeries::GetColorRepeating(int) const", referenced from:
      pcl::visualization::PCLPlotter::addPlotData(double const*, double const*, unsigned long, char const*, int, char const*) in pcl_plotter.cpp.o
```

relevant CMake output:
```
[…[
-- VTK version: 8.2.0
-- VTK rendering backend: OpenGL2
-- VTK Qt support: YES
-- VTK include: /usr/local/opt/vtk@8.2/include/vtk-8.2;/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9;/usr/local/include;/usr/local/opt/hdf5/include;/usr/local/opt/szip/include
-- VTK libs: vtksys;vtkCommonCore;vtkCommonMath;vtkCommonMisc;vtkCommonSystem;vtkCommonTransforms;vtkCommonDataModel;vtkCommonColor;vtkCommonExecutionModel;vtkCommonComputationalGeometry;vtkFiltersCore;vtkFiltersGeneral;vtkImagingCore;vtkImagingFourier;vtkFiltersStatistics;vtkFiltersExtraction;vtkInfovisCore;vtkFiltersGeometry;vtkFiltersSources;vtkRenderingCore;/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libz.tbd;vtkfreetype;vtkRenderingFreeType;vtkRenderingContext2D;vtkChartsCore;vtkDICOMParser;vtkdoubleconversion;vtklz4;vtklzma;vtkIOCore;vtkIOLegacy;/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libexpat.tbd;vtkIOXMLParser;vtkDomainsChemistry;vtkglew;vtkRenderingOpenGL2;vtkDomainsChemistryOpenGL2;vtkIOXML;vtkParallelCore;vtkFiltersAMR;vtkFiltersFlowPaths;vtkFiltersGeneric;vtkImagingSources;vtkFiltersHybrid;vtkFiltersHyperTree;vtkImagingGeneral;vtkFiltersImaging;vtkFiltersModeling;vtkFiltersParallel;vtkFiltersParallelImaging;vtkFiltersPoints;vtkFiltersProgrammable;/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/lib/libpython3.9.dylib;vtkPythonInterpreter;vtkWrappingTools;vtkWrappingPythonCore;vtkFiltersPython;vtkFiltersSMP;vtkFiltersSelection;vtkFiltersTexture;vtkFiltersTopology;verdict;vtkFiltersVerdict;vtkInteractionStyle;vtkGUISupportQt;vtksqlite;vtkIOSQL;vtkGUISupportQtSQL;vtkmetaio;/usr/local/lib/libjpeg.dylib;/usr/local/lib/libpng.dylib;/usr/local/lib/libtiff.dylib;vtkIOImage;vtkImagingHybrid;vtkInfovisLayout;vtkImagingColor;vtkRenderingAnnotation;vtkRenderingVolume;vtkInteractionWidgets;vtkViewsCore;vtklibproj;vtkGeovisCore;/usr/local/lib/libhdf5.dylib;/usr/local/lib/libsz.dylib;/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libdl.tbd;/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libm.tbd;/usr/local/lib/libhdf5_hl.dylib;vtkIOAMR;vtkIOAsynchronous;vtkpugixml;vtkIOCityGML;vtkIOEnSight;/usr/local/lib/libnetcdf.dylib;vtkexodusII;vtkIOExodus;vtkgl2ps;vtkRenderingGL2PSOpenGL2;vtkIOExport;vtkIOExportOpenGL2;vtklibharu;vtkIOExportPDF;vtkIOGeometry;vtkIOImport;/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libxml2.tbd;vtkIOInfovis;vtkIOLSDyna;vtkIOMINC;vtkogg;vtktheora;vtkIOMovie;vtkIONetCDF;vtkIOPLY;vtkjsoncpp;vtkIOParallel;vtkIOParallelXML;vtkIOSegY;vtkIOTecplotTable;vtkIOVeraOut;vtkIOVideo;vtkImagingMath;vtkImagingMorphological;vtkImagingStatistics;vtkImagingStencil;vtkInfovisBoostGraphAlgorithms;vtkInteractionImage;vtkPythonContext2D;vtkRenderingContextOpenGL2;vtkRenderingFreeTypeFontConfig;vtkRenderingImage;vtkRenderingLOD;vtkRenderingLabel;vtkRenderingQt;vtkRenderingVolumeOpenGL2;vtkViewsContext2D;vtkViewsInfovis;vtkViewsQt
[…]
```

**Comment by themightyoarfish** [comment]
There are some libs missing I think. This diff makes it link

```
diff --git a/visualization/CMakeLists.txt b/visualization/CMakeLists.txt
index d5ea61953..32737d052 100644
--- a/visualization/CMakeLists.txt
+++ b/visualization/CMakeLists.txt
@@ -197,6 +197,8 @@ else()
   target_include_directories("${LIB_NAME}" SYSTEM PUBLIC ${VTK_INCLUDE_DIRS})
   target_link_libraries("${LIB_NAME}"
                         vtkChartsCore
+                       vtkCommonColor
+                       vtkCommonComputationalGeometry
                         vtkFiltersExtraction
                         vtkFiltersGeometry
                         vtkFiltersModeling
```

**Comment by mvieth** [comment]
Thanks for testing! I added the two libraries (wrapped in an `if` so they are only linked for VTK 8). I wonder why this problem didn't appear when I tested under Ubuntu

File: tools/CMakeLists.txt [hunk 1]
```diff
@@ -203,6 +203,10 @@ else()
 
   if(${VTK_VERSION} VERSION_GREATER_EQUAL 9.0)
     target_link_libraries(pcl_obj2pcd VTK::FiltersCore)
+  else()
```
Comment (larshg) [inline tools/CMakeLists.txt:206]: These two could just be elseif(cond) ?

Comment (mvieth) [reply tools/CMakeLists.txt]: True. I updated it

**Review by larshg — APPROVED** [review]

**Pull request merged**